apps/pipeline-visualization/src/mermaid_builder.ts

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+import { strict as assert } from 'assert';
+
+import {
+  BlockDefinition,
+  PipelineDefinition,
+} from '@jvalue/jayvee-language-server';
+
+export interface MermaidOptions {
+  mermaidFile: string;
+  styleFile: string;
+  compositeBlocks: boolean;
+  properties: boolean;
+}
+
+export class MermaidBuilder {
+  protected linkLines: string[] = [];
+  protected nodeLines: string[] = [];
+  protected styleLines: string[] = [];
+  protected indent = 0;
+
+  constructor(protected options: MermaidOptions) {}
+
+  escapeHtml(unsafe: string) {
+    return unsafe
+      .replace(/&/g, '&amp;')
+      .replace(/</g, '&lt;')
+      .replace(/>/g, '&gt;')
+      .replace(/"/g, '&quot;')
+      .replace(/'/g, '&#039;');
+  }
+
+  diagram(diagramType: string, diagramDirection: string) {
```

Comment:
(Optional improvement / if you want to flex with your TypeScript Skills) Use [literal types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types).

E.g. `diagram(diagramType: string, diagramDirection: 'TD' | 'LR') {` (Use the values mermaid understands).

Also, be aware that implementation details are leaked here. However, in this case, I'm fine with that (since these are rather generic instructors that just happen to be mermaid syntax).